src/App.js

```diff
@@ -1,16 +1,43 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 import chatMessages from './data/messages.json';
+import ChatLog from './components/ChatLog';
 
 const App = () => {
+  const [messages, setMessages] = useState(chatMessages);
+  const [likeCount, setLikeCount] = useState(0);
```

Comment:
While it does work to add `likeCount` to state and pass the update function down as a prop to the `ChatEntry` component to be called on button click, there isn't a need to do so. Because of the way state updates work, whenever a state update function is called, the component is re-rendered and therefore every piece of JS in the component is also executed. This means that you can compute the like count based on the updated messages either by filtering like
```js
const totalLikes = messages.filter(message => message.liked).length;
```
or like
```js
const totalLikes = messages.reduce((message, total) => {
  if (message.liked) {
    total += 1;
  }
  return total;
}, 0);
```

And indeed there's a good reason not to update the like count using state. Because every time an update function is called, the component re-renders, so calling two update functions as you're doing when someone clicks the like button, will cause 2 re-renders. At this small scale, that's not very expensive, but re-rendering a lot of components can start to cause UI lag, which you'll want to avoid.